gradio/utils.py

```diff
@@ -1424,3 +1425,34 @@ def error_payload(
         content["duration"] = error.duration
         content["visible"] = error.visible
     return content
+
+
+class UnhashableKeyDict(MutableMapping):
+    def __init__(self):
+        self.data = []
+
+    def __getitem__(self, key):
+        for k, v in self.data:
+            if k == key:
+                return v
+        raise KeyError(key)
+
+    def __setitem__(self, key, value):
+        for i, (k, _) in enumerate(self.data):
+            if k == key:
```

Comment:
I'm not sure this equality will work for pandas/numpy types that are allowed in examples. Usually throws an error when you do a naive comparison.